clang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -2591,11 +2591,15 @@ static bool addCachedModuleFileToInMemoryCache(
     if (!Value) {
       Diag << Value.takeError();
     } else {
-      std::string ErrStr("no such entry in action cache; expected compile:\n");
+      std::string ErrStr("module file is not available in the CAS; expected to "
```

Comment:
I wonder if we could instead change this error to match what we do in the ASTReader:

```
          Diag(diag::err_ast_file_not_found)
              << moduleKindForDiagnostic(ImportedKind) << ImportedFile << true
              << ("missing or unloadable module cache key" + ImportedCacheKey)
                     .str();
```

which is something like
> module file '...pcm' not found: missing or unloadable module cache key llvmcas://....

I guess in this case we know if it's "unloadable" or if it's "missing" so we could be more precise and then for missing we could still dump the key.

Not sure if printing the whole pcm file path is helpful or if we should only print the basename, but to me this feels like a somewhat nicer kind of error.  We could still dump the key at the end